scripts/ledger.ts

```diff
@@ -35,7 +34,12 @@ export class LedgerNodeWallet implements Wallet {
     return new LedgerNodeWallet(derivationPath, transport, publicKey);
   }
 
-  async signTransaction(transaction: Transaction): Promise<Transaction> {
+  async signTransaction<T extends Transaction | VersionedTransaction>(
+    transaction: T
+  ): Promise<T> {
+    if (transaction instanceof VersionedTransaction) {
+      throw new Error("VersionedTransaction not supported by Ledger");
```

Comment:
check https://github.com/fogo-foundation/fogo-sessions/blob/main/scripts/ts/src/ledger.ts#L127 for a mor e updated version that supports VersionedTransaction too